lib/src/controls/form/number_box.dart

```diff
@@ -0,0 +1,412 @@
+  none,
+}
+
+/// A fluent design input form for numbers.
+///
+/// A NumberBox lets the user enter a number. If the user input a wrong value
+/// (a NaN value), the previous valid value is used.
+///
+///
+/// The value can be changed in several ways:
+///   - by input a new value in the text field
+///   - with increment/decrement buttons (only with modes
+///     [SpinButtonPlacementMode.inline] or [SpinButtonPlacementMode.compact]).
+///   - by use the wheel scroll on the number box when he have the focus
+///   - with the shortcut [LogicalKeyboardKey.pageUp] and
+///     [LogicalKeyboardKey.pageDown].
+///
+/// Modes:
+///  [SpinButtonPlacementMode.inline] : Show two icons as a suffix of the text
+///  field. With for increment the value and one for decrement the value.
+///  [SpinButtonPlacementMode.compact] : Without the focus, it's appears like
+///  a normal text field. But when the widget has the focus, an overlay is
+///  visible with a button for increment the value and another for decrement
+///  the value.
+///  [SpinButtonPlacementMode.none] : Don't show any additional button on the
+///  text field.
+///
+/// If the parameter [clearButton] is enabled, an additional icon is shown
+/// for clear the value when the widget has the focus.
+///
+/// See also:
+///
+///  * https://learn.microsoft.com/en-us/windows/apps/design/controls/number-box
+class NumberBox extends StatefulWidget {
+  /// The value of the number box. When this value is null, the number box field
+  /// is empty.
+  final int? value;
+
+  /// Called when the value of the number box change.
+  /// The callback is fired only if the user click on a button or the focus is
+  /// lost.
+  ///
+  /// If the [onChanged] callback is null then the number box widget will
+  /// be disabled, i.e. its buttons will be displayed in grey and it will not
+  /// respond to input.
+  final ValueChanged<int?>? onChanged;
+
+  /// {@macro flutter.widgets.Focus.focusNode}
+  final FocusNode? focusNode;
+
+  /// Display modes for the Number Box.
+  final SpinButtonPlacementMode mode;
+
+  /// When false, it disable the suffix button with a cross for remove the
+  /// content of the number box.
+  final bool clearButton;
+
+  /// The value that is incremented or decremented when the user click on the
+  /// buttons or when he scroll on the number box.
+  final int smallChange;
+
+  /// The value that is incremented when the user click on the shortcut
+  /// [LogicalKeyboardKey.pageUp] and decremented when the user lick on the
+  /// shortcut [LogicalKeyboardKey.pageDown].
+  final int largeChange;
+
+  const NumberBox({
+    super.key,
+    required this.value,
+    required this.onChanged,
+    this.focusNode,
+    this.mode = SpinButtonPlacementMode.compact,
+    this.clearButton = true,
+    this.smallChange = 1,
+    this.largeChange = 10,
+  });
+
+  @override
+  State<StatefulWidget> createState() => _NumberBoxState();
+}
+
+class _NumberBoxState extends State<NumberBox> {
```

Comment:
maybe exposing the state class as public and the increment functions, so that it is possible to manipulate it with a `GlobalKey<NumberBoxState>`?

Reply:
I'm sorry but I don't known how to do that. Feel free to add a commit for doing that ;)

Reply:
Just remove the _ (underscore) from the state class name

Reply:
Oh I see, that was the globalKey that I didn’t understand. But I see now. I will do that tomorrow